### ISSUE TITLE
Feature request: possibility to programatically focus/blur all components

### ISSUE THREAD

**Issue by smashercosmo** [jquense/react-widgets#339]
Currently Multiselect is the only component that provides direct access to its focus method. For other components you have to do ugly things like 
```
ReactDOM.findDOMNode(this.refs.combobox.refs.inner.refs.input).focus();
```
And also all components lack corresponding blur() method. Would be nice to have them. If you are ok with that, I can provide PR.

**Comment by jquense** [comment]
happy to take a PR adding consistent focus method for each widget, I'm not quite sure what the utility of the `blur()` method would be tho, generally removing focus from the document is a bad practice, and focusing something else would remove focus from the widgets anyway.

practical point: most widgets already have a `focus()` method but don't tell `uncontrollable` to surface it on, you can see how the API works from the multiselect: https://github.com/jquense/react-widgets/blob/master/src/Multiselect.jsx#L519

**Comment by smashercosmo** [comment]
> practical point: most widgets already have a focus() method but don't tell uncontrollable
yep, I've already dug through the code)

**Comment by smashercosmo** [comment]
> I'm not quite sure what the utility of the blur()

well, for example our use case is the following:
we use Multiselect as a search input and on mobile devices it's pretty annoying for users to have keyboard opened, which overlaps search results. So we need to blur component after adding every new tag to Multiselect to close it.

**Comment by jquense** [comment]
wouldn't it make more sense to `preventDefault()` on focus in those cases though?

**Comment by smashercosmo** [comment]
Didn't get it.
1. User clicks Multiselect
2. Multiselect becomes focused
3. Mobile keyboard opens
3. User adds new tag
4. Now we need to close keyboard, so user can see search results
When do we need to call preventDefault()?

**Comment by smashercosmo** [comment]
So, to blur or not to blur?)) I need to know that before starting working on PR

**Comment by jquense** [comment]
lets go without the blur for now, and maybe open another issue for dealing with the keyboard issue...manual blur may be the right way to fix that but I'd like to spend more time thinking about it before adding another api surface

**Comment by jquense** [comment]
closed by #344

**Comment by luisfuertes** [comment]
It is already fixed? How can i do blur after select date in DateTimePicker?
